hkdf/src/hkdf.rs

```diff
@@ -64,18 +69,21 @@ impl<D> Hkdf<D>
 
             let output = output_block.result().code();
             let needed = cmp::min(remaining, hmac_output_bytes);
-            okm.extend(&output[..needed]);
+
+            okm[offset..offset+needed].copy_from_slice(&output[..needed]);
+            offset += needed;
+
             prev = Some(output);
             blocknum += 1;
             remaining -= needed;
         }
-
-        okm
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use ::std::vec::Vec;
+
```

Comment:
I think we may be better just move tests out into a separate `tests/` directory. This will allow use to remove `std` notion from the main code.